ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -669,6 +763,22 @@ private void checkForDeviceDimensionsChanges() {
       }
     }
 
+    private void checkForLayoutContextChanges() {
+      RootViewInsets safeAreaInsets = getSafeAreaInsets();
+      if (mFrameWidth != getWidth() ||
+        mFrameHeight != getHeight() ||
+        mFrameX != getX() ||
+        mFrameY != getY() ||
+        !areInsetsEqual(mSafeAreaInsets, safeAreaInsets)) {
+        mSafeAreaInsets = safeAreaInsets;
+        mFrameWidth = getWidth();
+        mFrameHeight = getHeight();
+        mFrameX = getX();
+        mFrameY = getY();
+        emitLayoutContextEvent(safeAreaInsets);
```

Comment:
Will this trigger if the insets change when the app is drawn below the statusbar/navigation bar? Maybe we need a `setOnApplyWindowInsetsListener`?

Reply:
I think it did work, I remember testing different combinations of toggling the status bar hidden and translucent props. Seems like the global layout listener does get called when system insets change.